pyrit/orchestrator/multi_turn/tree_of_attacks_with_pruning_orchestrator.py

```diff
@@ -0,0 +1,240 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+import random
+
+from pathlib import Path
+from treelib import Tree
+from typing import Optional
+
+from pyrit.common.path import DATASETS_PATH
+from pyrit.memory import MemoryInterface
+from pyrit.models import SeedPrompt
+from pyrit.orchestrator import MultiTurnOrchestrator, MultiTurnAttackResult
+from pyrit.prompt_converter import PromptConverter
+from pyrit.prompt_target import PromptTarget, PromptChatTarget
+from pyrit.score import SelfAskTrueFalseScorer, SelfAskScaleScorer, TrueFalseQuestion
+from pyrit.score.scorer import Scorer
+
+from pyrit.orchestrator.multi_turn.tree_of_attack_node import TreeOfAttackNode
+
+logger = logging.getLogger(__name__)
+
+
+class TapAttackResult(MultiTurnAttackResult):
```

Comment:
```suggestion
class TAPAttackResult(MultiTurnAttackResult):
```
nit: it's an acronym. Also, "tap" (not all caps) is an actual word. 
Or alternatively, let's spell it out?